Saule/Serialization/ResourceSerializer.cs

```diff
@@ -130,7 +130,10 @@ private JToken CreateTopLevelLinks(int count)
         {
             var result = new JObject();
 
-            if (_resource.LinkType.HasFlag(LinkType.Self))
+            // if resource has Top only and not Self, then we render it.
+            // otherwise to preserve back compatibility if Self is enabled, then we also render it
+            if ((_resource.LinkType.HasFlag(LinkType.Top) && !_resource.LinkType.HasFlag(LinkType.Self))
```

Comment:
I think this can be simplified to `Top || Self`?

Reply:
Agree. Updated it